.github/actions/install-deps/action.yaml

```diff
@@ -7,14 +7,14 @@ inputs:
 runs:
   using: "composite"
   steps:
-    - uses: actions/setup-go@v4
-      with:
-        go-version-file: go.mod
-        check-latest: true
-        cache-dependency-path: "**/go.sum"
     # Root path permission workaround for caching https://github.com/actions/cache/issues/845#issuecomment-1252594999
     - run: sudo chown "$USER" /usr/local
       shell: bash
+    - uses: actions/setup-go@v4
```

Comment:
Why is this moved? Also, changing this `go.mod` file is going to result in some odd behavior and some potential conflicts if the `hack/go.mod` version doesn't match with the regular `go.mod` version